packages/compass-schema/src/components/export-schema/export-schema-modal.tsx

```diff
@@ -0,0 +1,321 @@
+
+import { JSONEditor } from '@mongodb-js/compass-editor';
+import { ActionButton } from './export-schema-modal-action-button';
+import { exportMongodbJSONSchema } from './formats/mongodb-json-schema';
+import { exportCompassInternalSchema } from './formats/compass-internal-schema';
+import { exportJSONSchema } from './formats/ejson-json-schema';
+
+type ExportSchemaModalProps = {
+  schema: Schema;
+  closeExportSchema: () => void;
+  exportSchemaOpened: boolean;
+};
+
+type Export = { code?: string; error?: string };
+
+type ExportPanelProps = { settings?: JSX.Element; exportInfo: Export };
+
+const exportPanelStyles = css({
+  display: 'grid',
+  gridTemplateColumns: `1fr 1fr`,
+  gridTemplateRows: '1fr',
+  gap: spacing[2],
+  maxHeight: '100%',
+  position: 'relative',
+});
+
+const exportPanelCodeAreaStyles = css({
+  position: 'relative',
+});
+
+const editorStyles = css({
+  width: spacing[6] * 10,
+  height: spacing[6] * 7,
+});
+
+const exportPanelCopyButtonStyles = css({
+  position: 'absolute',
+  top: spacing[2],
+  right: spacing[3],
+});
+
+const exportPanelSettingsAreaStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[2],
+});
+
+const ExportPanel: React.FunctionComponent<ExportPanelProps> = ({
+  settings,
+  exportInfo,
+}) => {
+  const handleCopy = useCallback(() => {
+    if (exportInfo.code) {
+      void navigator.clipboard.writeText(exportInfo.code);
+      return true;
+    }
+
+    return false;
+  }, [exportInfo.code]);
+
+  return (
+    <div className={exportPanelStyles}>
+      <div className={exportPanelCodeAreaStyles}>
+        {exportInfo.error ? (
+          <Banner variant={BannerVariant.Danger}>
+            Error: {exportInfo.error}
+          </Banner>
+        ) : (
+          <>
+            <WorkspaceContainer className={editorStyles}>
+              <JSONEditor
+                text={exportInfo.code || ''}
+                readOnly={true}
+                showFoldGutter={false}
+                showLineNumbers={false}
+              />
+            </WorkspaceContainer>
+            <div className={exportPanelCopyButtonStyles}>
+              <ActionButton
+                key="Copy"
+                label="Copy"
+                icon="Copy"
+                onClick={handleCopy}
+              />
+            </div>
+          </>
+        )}
+      </div>
+
+      {settings && (
+        <div className={exportPanelSettingsAreaStyles}>{settings}</div>
+      )}
+    </div>
+  );
+};
+
+function useSchemaConversion<T>(
+  schema: Schema,
+  settings: T,
+  converter: (schema: Schema, settings: T) => string
+): Export {
+  const exportInfo = useMemo(() => {
+    try {
+      return {
+        code: converter(schema, settings),
+      };
+    } catch (e) {
+      const errorMessage = (e as Error)?.message || '';
+      return { error: `Conversion failed: ${errorMessage}` };
+    }
+  }, [converter, schema, ...(settings && Object.values(settings))]);
```

Comment:
```suggestion
  }, [converter, schema, ...(settings && Object.entries(settings).flat())]);
```

Otherwise `{ a: 1 }` and `{ b: 1 }` are considered to be the same, right?

Reply:
Good catch!